diff --git a/loadgen/CMakeLists.txt b/loadgen/CMakeLists.txt
index 9649c96..69b59a6 100644
--- a/loadgen/CMakeLists.txt
+++ b/loadgen/CMakeLists.txt
@@ -8,7 +8,7 @@ set(mlperf_loadgen_VERSION_MINOR 5)
 message("mlperf_loadgen v${mlperf_loadgen_VERSION_MAJOR}.${mlperf_loadgen_VERSION_MINOR}")
 
 # Set build options.
-set(CMAKE_CXX_FLAGS "-O2 -W -Wall")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2 -W -Wall")
 message(STATUS "Using compiler flags: ${CMAKE_CXX_FLAGS}")
 message(STATUS "Using static linker flags: ${CMAKE_STATIC_LINKER_FLAGS}")
 message(STATUS "Using shared linker flags: ${CMAKE_SHARED_LINKER_FLAGS}")
